models/avatar.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"crypto/md5"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"code.gitea.io/gitea/modules/cache"
+	"code.gitea.io/gitea/modules/setting"
+)
+
+var ()
+
+// EmailHash represents a pre-generated hash map
+type EmailHash struct {
+	ID     int64  `xorm:"pk autoincr"`
```

Comment:
No need for id column, md5sum can be used as primary key column. Also it would probably be better to name it more generic ex. `Hash`